osu.Game/Updater/UpdateManager.cs

```diff
@@ -35,24 +37,37 @@ protected override void LoadComplete()
         {
             base.LoadComplete();
 
+            Schedule(() => Task.Run(CheckForUpdateAsync));
+
+            // debug / local compilations will reset to a non-release string.
+            // can be useful to check when an install has transitioned between release and otherwise (see OsuConfigManager's migrations).
+            config.Set(OsuSetting.Version, game.Version);
+        }
+
+        public async Task CheckForUpdateAsync()
+        {
+            if (!CanCheckForUpdate)
+                return;
```

Comment:
No, don't silently return when you shouldn't be called, throw `NotSupportedException` here.

Reply:
reasoning?

Reply:
`CheckForUpdateAsync` shouldn't be called if the `UpdateManager` itself isn't capable of doing it, that should be identified as a bug.

Reply:
I don't follow. It's a public method. Are you expecting every usage to first check the bool? That sounds like really bad API. Why don't we just make UpdateManager null and throw a nullref then? Isn't the point of making it abstract to simplify external implementation?

Reply:
Right, you have a point there. It's fine in this way then.

Reply:
> I don't follow. It's a public method. Are you expecting every usage to first check the bool? That sounds like really bad API. Why don't we just make UpdateManager null and throw a nullref then? Isn't the point of making it abstract to simplify external implementation?

I thought about doing that too, but UpdateManager seems to go beyond of just checking updates.

At this point I would suggest splitting UpdateManager into 2 classes for their own functionality, that being a class for checking for new updates, and another one for doing the startup work when a new update has been applied.

This way we can stick to abstracting the checking process and make it null on not deployed builds/mobile versions.

Reply:
I can't see the gain of splitting the two things. Sounds like extra complexity for not much gain.